store/src/java/com/zimbra/cs/mailbox/MailSender.java

```diff
@@ -494,6 +536,13 @@ public ItemId sendMimeMessage(OperationContext octxt, Mailbox mbox, Boolean save
            return sendMimeMessage(octxt, mbox, saveToSent, mm, uploads, origMsgId, replyType, identity, replyToSender, null);
     }
 
+    public ItemId sendMimeMessage(OperationContext octxt, Mailbox mbox, Boolean saveToSent, MimeMessage mm,
+            Collection<Upload> uploads, ItemId origMsgId, String replyType, Identity identity, boolean replyToSender,
+            MimeProcessor mimeProc, boolean deliveryReport) throws ServiceException {
+        mDeliveryReport = deliveryReport;
+        return sendMimeMessage(octxt, mbox, saveToSent, mm, uploads, origMsgId, replyType, identity, replyToSender, mimeProc);
+    }
```

Comment:
refactoring at lines 520-544: execute all the process in the 3rd `sendMimeMessage` defined at 539. 
2nd (at line 533) calls the 1st (at line 520). 1st calls the 3rd.

Reply:
1st doesn’t calls the `3rd` rather `4th` method `sendMimeMessage(OperationContext octxt, Mailbox mbox, MimeMessage mm)` at line `549`

Reply:
My idea:
```
(From line 517)
    /**
     * Sets member variables and sends the message.
     */
    public ItemId sendMimeMessage(OperationContext octxt, Mailbox mbox, Boolean saveToSent, MimeMessage mm,
            Collection<Upload> uploads, ItemId origMsgId, String replyType, Identity identity, boolean replyToSender, MimeProcessor mimeProc)
            throws ServiceException {
        return sendMimeMessage(octxt, mbox, saveToSent, mm, uploads, origMsgId, replyType, identity, replyToSender, mimeProc, false); // Call the 3rd sendMimeMessage
    }

    public ItemId sendMimeMessage(OperationContext octxt, Mailbox mbox, Boolean saveToSent, MimeMessage mm,
            Collection<Upload> uploads, ItemId origMsgId, String replyType, Identity identity, boolean replyToSender)
            throws ServiceException {
           return sendMimeMessage(octxt, mbox, saveToSent, mm, uploads, origMsgId, replyType, identity, replyToSender, null);  // No change. Call the 1st sendMimeMessage
    }

    public ItemId sendMimeMessage(OperationContext octxt, Mailbox mbox, Boolean saveToSent, MimeMessage mm,
            Collection<Upload> uploads, ItemId origMsgId, String replyType, Identity identity, boolean replyToSender,
            MimeProcessor mimeProc, boolean deliveryReport) throws ServiceException {
        mSaveToSent = saveToSent;     // Move existing code from the 1st to here
        mUploads = uploads;
        mOriginalMessageId = origMsgId;
        mReplyType = replyType;
        mIdentity = identity;
        mReplyToSender = replyToSender;
        mimeProcessor = mimeProc;
        mDeliveryReport = deliveryReport;   // Add new process as you did

        return sendMimeMessage(octxt, mbox, mm);  // Move existing code from the 1st 
    }
```

Reply:
Done